src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -370,7 +370,10 @@ template<
 >
 void GAN<Model, InitializationRuleType, Noise, PolicyType>::Shuffle()
 {
-  math::ShuffleData(predictors, responses, predictors, responses);
```

Comment:
Are you sure this doesn't work? It would rather indicate an issue with the `math::ShuffleData` function.